### ISSUE TITLE
Caught exception: TypeError: Cannot read property 'substring' of undefined

### ISSUE THREAD

**Issue by michaelzangl** [zkoss/zkless-engine#3]
Version 0.1.22
Since this seems to be the only one for which there is documentation on how to run it:
https://www.zkoss.org/wiki/ZK_Style_Customization_Guide/Integrate_with_LESS/How_ZK_works_with_LESS/Compile_LESS#Compile_LESS_by_Ant

Problem is this code (typical var scope bug)
```js
				if (cf.includes('web')) {
					var tempPath = genDir + cf.substring(cf.indexOf('web'));
				}
				mkpath.sync(tempPath.substring(0, tempPath.lastIndexOf('/')), 0700);
```

Stack trace:
```
Caught exception: TypeError: Cannot read property 'substring' of undefined
    at encodeImports (/home/mzangl/workspace/riscage/base/node_modules/zkless-engine/lib/CpLess.js:133:26)
    at importCheck (/home/mzangl/workspace/riscage/base/node_modules/zkless-engine/lib/CpLess.js:117:4)
    at copyImportsFiles (/home/mzangl/workspace/riscage/base/node_modules/zkless-engine/lib/CpLess.js:111:7)
    at copyImports (/home/mzangl/workspace/riscage/base/node_modules/zkless-engine/lib/CpLess.js:60:5)
    at Walker.<anonymous> (/home/mzangl/workspace/riscage/base/node_modules/zkless-engine/lib/CpLess.js:49:5)
    at Walker.emit (events.js:198:13)
    at Walker._wNext (/home/mzangl/workspace/riscage/base/node_modules/walk/lib/walk.js:278:10)
    at Walker._wNext (/home/mzangl/workspace/riscage/base/node_modules/walk/lib/walk.js:273:17)
    at Walker._wOnEmitDone (/home/mzangl/workspace/riscage/base/node_modules/walk/lib/walk.js:125:8)
    at nextWhenReady (/home/mzangl/workspace/riscage/base/node_modules/walk/lib/node-type-emitter.js:72:29)
```

Workaround: Input folder may not be called .../webapp, but must be somewehre below a directory called .../web/...

**Comment by cor3000** [comment]
version 0.1.22 is an old version and superseeded by version >1, the documentation mentioned is also based on old ZK versions 7.0.3.

The run documentation for the latest version is in the README of this repo
-> https://github.com/zkoss/zkless-engine/blob/master/README.md

New custom themes should be built using the zktheme template (which integrates the zkless-engine 1.x):
https://github.com/zkoss/zkThemeTemplate

Currently there are no plans to maintain/fix the old version and the new one should be used.
Is there a specific task you are trying to achieve? (also which versions are involved)

**Comment by michaelzangl** [comment]
Thanks for the reply.

I tried to compile an old ZK theme (legacy software …) and migrate it to Gradle. The problem is, that the less files in ZK 8.6 JAR file contain some <%..%> tags and other expressions that were handled by the old version. I had problems using the new version there, it always complained about syntax errors on those tags.

The ZK docs are always hard to find - the current docs do not cover legacy functionality and in my experience, often the only solution is to read the source code and find out how it was once done / how to migrate.

I now fixed it for us by:
* Use gradle to extract the ZK Jar files
* While extracting, use some regex replacements to get rid of all the tags (we don't need them / can work around them)
* Then use the normal lessc and set the include path to the extracted less files.
This also allowed me to drop the .dsp-Filter, too.

> In case you are a customer please contact our support.
I know, but we have a very specific setup (We are trying to drop ZK for years and replace it with something more maintainable, but for now, I am stuck with this and want to spend as few time changing things as possible …). Gradle does not even seem to be officially supported for theme compilation.
